storage/innobase/log/log0log.cc

```diff
@@ -975,19 +1059,18 @@ void log_write_up_to(lsn_t lsn, bool durable,
 #ifdef HAVE_PMEM
   if (log_sys.is_pmem())
   {
-    ut_ad(!callback);
     if (durable)
       log_sys.persist(lsn, false);
     return;
   }
 #endif
+  ut_ad(!log_sys.is_mmap());
 
 repeat:
   if (durable)
   {
     if (flush_lock.acquire(lsn, callback) != group_commit_lock::ACQUIRED)
       return;
-    flush_lock.set_pending(log_sys.get_lsn());
```

Comment:
seems like a random change. I tend to remove all "set_pendings" later, I doubt their usefulness, but maybe not in this unrelated patch.

Reply:
I must have been thinking that `log_flush()` would set the value a little later, after we’d have done some `pwrite()` and `fdatasync()`. Now I realize that the purpose of this early assignment is to ensure that concurrent waiters on `flush_lock` would be given an early promise that something is going to be made durable up to a particular LSN.

I will retain this code and add a comment.

Reply:
in theory this gave waiters chance to spinloop a bit, if the expected soon-to-be flush_lsn LSN is less than lsn parameter.
In practice though, no performance improvement was ever measured, and  this spin was never noticed by any profiler.